sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/ShowTblPropertiesSuite.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.command.v1
+
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.execution.command
+import org.apache.spark.sql.internal.SQLConf
+
+/**
+ * This base suite contains unified tests for the `SHOW TBLPROPERTIES` command that checks V1
+ * table catalogs. The tests that cannot run for all V1 catalogs are located in more
+ * specific test suites:
+ *
+ *   - V1 In-Memory catalog: `org.apache.spark.sql.execution.command.v1.ShowTblPropertiesSuite`
+ *   - V1 Hive External catalog:
+ *     `org.apache.spark.sql.hive.execution.command.ShowTblPropertiesSuite`
+ */
+trait ShowTblPropertiesSuiteBase extends command.ShowTblPropertiesSuiteBase
+    with command.TestsV1AndV2Commands {
+
+  test("SHOW TBLPROPERTIES(KEY) KEY NOT FOUND") {
+    withNamespaceAndTable("ns1", "tbl") { tbl =>
+      val nonExistingKey = "nonExistingKey"
+      spark.sql(s"CREATE TABLE $tbl (id bigint, data string) $defaultUsing " +
+        s"TBLPROPERTIES ('user'='andrew', 'status'='new')")
+
+      val properties = sql(s"SHOW TBLPROPERTIES $tbl ('$nonExistingKey')")
+      val expected = Seq(Row(nonExistingKey,
+        s"Table ns1.tbl does not have property: $nonExistingKey"))
```

Comment:
The table name without catalog name in V1 different with V2.

Reply:
can we change the test code so that it can be moved to the base class? e.g.
```
val res = properties.collect()
assert(res.length == 1)
assert(res.head.getString(0) == nonExistingKey)
assert(res.head.getString(1).contains("does not have property: $nonExistingKey"))
```